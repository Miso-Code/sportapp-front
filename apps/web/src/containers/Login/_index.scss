.login-form {
	@apply bg-white p-4 rounded-lg shadow-md flex flex-col items-center gap-4 w-full;

	min-width: 300px;
	max-width: 466px;
}
