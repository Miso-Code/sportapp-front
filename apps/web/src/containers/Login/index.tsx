import TextFieldController from 'components/Inputs/TexFieldController'
export default function LoginContainer() {
	return (
		<form>
			<TextFieldController />
		</form>
	)
}
