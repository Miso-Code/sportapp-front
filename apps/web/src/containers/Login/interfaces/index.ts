import { FormData } from 'containers/Login/utils/schema'

export interface Props {
	onHandleSubmit: (data: FormData) => void
	isDisabled?: boolean
}
