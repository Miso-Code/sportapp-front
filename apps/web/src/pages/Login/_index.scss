.login {
	@apply flex;

	min-width: 100vw;
	min-height: 100vh;

	.section-main {
		@apply flex-1 p-2 flex flex-col items-center justify-center;
	}

	.section-secondary {
		@apply w-1/2 lg:block hidden;
	}

	.container-image {
		@apply w-full h-full;
	}

	.image-login {
		@apply object-cover object-center w-full h-full;
	}
}
