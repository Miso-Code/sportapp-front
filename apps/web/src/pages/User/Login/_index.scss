.login {
	@apply flex;

	min-height: 100vh;

	.title {
		@apply font-bold mb-7 text-6xl md:text-8xl;
	}

	.section {
		&-main {
			@apply flex-1 p-2 flex flex-col items-center justify-center pb-10;
		}
	}

	.navigation {
		@apply mt-4;
	}

	.card {
		&-login {
			@apply w-full p-7 pb-4 border;

			min-width: 18.75rem;
			max-width: 29.125rem;

			&__full {
				max-width: 44.375rem;
			}
		}

		&-title {
			@apply w-full text-center mb-6;
		}
	}
}

.alert-register-container {
	@apply absolute top-0 left-0 w-full;
}
