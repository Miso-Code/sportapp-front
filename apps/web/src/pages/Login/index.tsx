import SecondarySection from '@/components/SecondarySection'
import TransitionAlert from '@/components/TransitionAlert'
import LoginContainer from '@/containers/Login'
import { FormData } from '@/containers/Login/utils/schema'
import { Button, Paper, Typography } from '@mui/material'
import { useAuthStore } from '@sportapp/stores/src/auth'
import registerImage from 'assets/images/login-wallpaper.jpg'
import { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import './_index.scss'

export default function Login() {
	const [alert, setAlert] = useState(false)
	const navigate = useNavigate()
	const { t } = useTranslation()
	const { loading, error } = useAuthStore()
	const { login } = useAuthStore()

	const handleSubmit = async (data: FormData) => {
		const result = await login(data)

		if (result) {
			navigate('/home')
		} else {
			setAlert(true)
		}
	}

	const handleGoToRegister = () => {
		navigate('/register')
	}

	return (
		<>
			<div className='login'>
				<main className='section-main'>
					<Typography className='title' variant='h1'>
						{t('app.name')}
					</Typography>
					<Paper variant='outlined' className='card-login'>
						<Typography className='card-title' variant='h6'>
							{t('login.default')}
						</Typography>
						<LoginContainer
							isDisabled={loading}
							onHandleSubmit={handleSubmit}
						/>

						<Button
							type='button'
							fullWidth
							disabled={loading}
							onClick={handleGoToRegister}
							variant='text'
							className='navigation'
							title={t('login.button')}>
							{t('login.register.go')}
						</Button>
					</Paper>
				</main>
				<SecondarySection
					image={registerImage}
					altImage={t('login.image')}
				/>
			</div>
			<TransitionAlert
				containerClassName='alert-register-container'
				isOpen={alert}
				handleClose={setAlert}
				message={t(error ?? 'errors.register.base')}
				severity='error'
			/>
		</>
	)
}
