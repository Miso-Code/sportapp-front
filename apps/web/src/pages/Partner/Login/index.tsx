import SecondarySection from '@/components/SecondarySection'
import TransitionAlert from '@/components/TransitionAlert'
import LoginContainer from '@/containers/Login'
import { FormData } from '@/containers/Login/utils/schema'
import { Button, Paper, Typography } from '@mui/material'
import { usePartnerAuthStore } from '@sportapp/stores/src/partner/auth'
import registerImage from 'assets/images/login-partner-wallpaper.jpg'
import { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import './_index.scss'

export default function LoginPartner() {
	const [alert, setAlert] = useState(false)
	const navigate = useNavigate()
	const { t } = useTranslation()
	const { loading, error, logout } = usePartnerAuthStore()
	const { login } = usePartnerAuthStore()

	const handleSubmit = async (data: FormData) => {
		const payload = {
			email: data.email.toLowerCase(),
			password: data.password
		}
		const result = await login(payload)

		if (result) {
			navigate('/partner/home')
		} else {
			setAlert(true)
		}
	}

	const handleGoToRegister = () => {
		navigate('/partner/register')
	}

	useEffect(() => {
		logout()
	}, [logout])

	return (
		<>
			<div className='login-partner'>
				<main className='section-main'>
					<Typography className='title' variant='h1'>
						{t('app.name')}
					</Typography>
					<Paper variant='outlined' className='card-login-partner'>
						<Typography className='card-title' variant='h6'>
							{t('login.partner.title')}
						</Typography>
						<LoginContainer
							isDisabled={loading}
							onHandleSubmit={handleSubmit}
						/>

						<Button
							type='button'
							id='go-to-register'
							fullWidth
							disabled={loading}
							onClick={handleGoToRegister}
							variant='text'
							className='navigation-login-partner-register'
							title={t('login.button')}>
							{t('login.register.go')}
						</Button>
					</Paper>
				</main>
				<SecondarySection
					image={registerImage}
					altImage={t('login.image')}
				/>
			</div>
			<TransitionAlert
				containerClassName='alert-partner-login-container'
				isOpen={alert}
				handleClose={setAlert}
				message={t(error ?? 'errors.register.base')}
				severity='error'
			/>
		</>
	)
}
