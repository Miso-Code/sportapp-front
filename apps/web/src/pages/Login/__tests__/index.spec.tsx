import { render, RenderResult } from '@testing-library/react'
import Login from 'pages/Login'

describe('Login', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(<Login />)
	})

	afterEach(() => {
		wrapper.unmount()
	})

	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})
})
