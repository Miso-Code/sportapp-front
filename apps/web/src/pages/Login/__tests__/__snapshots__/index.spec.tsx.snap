// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login"
      >
        <main
          class="section-main"
        >
          <form>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Outlined
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-basic"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Outlined
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </form>
        </main>
        <section
          class="section-secondary"
        >
          <figure
            class="container-image"
          >
            <img
              alt="Login"
              class="image-login"
              src="mocked-file-content"
            />
          </figure>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login"
    >
      <main
        class="section-main"
      >
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Outlined
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Outlined
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </form>
      </main>
      <section
        class="section-secondary"
      >
        <figure
          class="container-image"
        >
          <img
            alt="Login"
            class="image-login"
            src="mocked-file-content"
          />
        </figure>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
